### PR TITLE
[SPARK-35385][SQL][TESTS] Skip duplicate queries in the TPCDS-related tests

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This PR proposes to skip the "q6", "q34", "q64", "q74", "q75", "q78" queries in the TPCDS-related tests because the TPCDS v2.7 queries have almost the same ones; the only differences in these queries are ORDER BY columns.

### Why are the changes needed?

To improve test performance.

### Does this PR introduce _any_ user-facing change?

No, dev only.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala [hunk 1]
```diff
@@ -36,6 +36,12 @@ trait TPCDSBase extends SharedSparkSession with TPCDSSchema {
     "q81", "q82", "q83", "q84", "q85", "q86", "q87", "q88", "q89", "q90",
     "q91", "q92", "q93", "q94", "q95", "q96", "q97", "q98", "q99")
 
+  // Since `tpcdsQueriesV2_7_0` has almost the same queries with these oens below,
+  // we skip them in the TPCDS-related tests.
+  private val excludedTpcdsQueries: Set[String] = Set("q6", "q34", "q64", "q74", "q75", "q78")
+
+  val tpcdsQueries: Seq[String] = tpcdsAllQueries.filterNot(excludedTpcdsQueries.contains)
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala:43]: another idea:
```
val tpcdsQueries: Seq[String] = 1.to(99).map("q" + _).flatMap { q => 
  if (Seq("q14", "q23", "q24", "q39").contains(q)) Seq(q + "a", q + "b") else Seq(q)
}.filterNot { q =>
  // ...
  Seq("q6", "q34", "q64", "q74", "q75", "q78").contains(q)
}
```

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala [hunk 2]
```diff
@@ -36,6 +36,12 @@ trait TPCDSBase extends SharedSparkSession with TPCDSSchema {
     "q81", "q82", "q83", "q84", "q85", "q86", "q87", "q88", "q89", "q90",
     "q91", "q92", "q93", "q94", "q95", "q96", "q97", "q98", "q99")
 
+  // Since `tpcdsQueriesV2_7_0` has almost the same queries with these oens below,
+  // we skip them in the TPCDS-related tests.
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala:40]: Well, this hides the previous reasoning. I believe it's worth to keep SPARK-35327 comment explicitly.
```
SPARK-35327: Filters out the TPC-DS queries that can cause flaky test results
```

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSQueryTestSuite.scala [hunk 3]
```diff
@@ -83,12 +83,6 @@ class TPCDSQueryTestSuite extends QueryTest with TPCDSBase with SQLQueryTestHelp
       .toFile.getAbsolutePath
   }
 
-  override val tpcdsQueries = {
-    // SPARK-35327: Filters out the TPC-DS queries that can cause flaky test results
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCDSQueryTestSuite.scala:87]: Please move this to the new location too.

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala (hunk 2 above)
Comment (maropu) [reply sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala:40]: Sure, udpated.

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala (hunk 1 above)
Comment (maropu) [reply sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala:45]: I feel the `if` part looks a bit tricky, so I prefer to keep it as it is.

**Comment by maropu** [comment]
The last commit is only for the comment update, so I merged to master. Thank you, @cloud-fan @dongjoon-hyun

**Pull request closed without merging**